tests/framework/clients/k8s_block.go

```diff
@@ -99,52 +103,59 @@ func (rb *BlockOperation) BlockMap(name string, mountpath string) (string, error
 //Output  - k8s exec pod operation output and/or error
 func (rb *BlockOperation) BlockWrite(name string, mountpath string, data string, filename string, namespace string) (string, error) {
 	wt := "echo \"" + data + "\">" + mountpath + "/" + filename
-	writeDataToBlockPod[2] = wt
-	option := []string{name}
+	args := []string{"exec", name}
+
 	if namespace != "" {
-		option = append(option, namespace)
+		args = append(args, "-n", namespace)
 	}
-	out, err, status := rb.transportClient.Execute(writeDataToBlockPod, option)
-	if status == 0 {
-		return out, nil
+	args = append(args, "--", "sh", "-c", wt)
+
+	result, err := rb.k8sClient.Kubectl(args...)
+	if err != nil {
+		return err.Error(), fmt.Errorf("Unable to write data to pod --: %s", err)
+
 	}
-	return err, fmt.Errorf("Unable to write data to pod: %s --> %s", err, out)
+	return result, nil
 
 }
 
 // BlockRead Function to read from block created by rook ,i.e. Read data from a pod that is using a pvc
-// Input paramaters -
+// Input parameters -
 //name - path to a yaml file that creates a pod  - pod should be defined to use a pvc that was created earlier
 //mountpath - folder on the pod were data is supposed to be written(should match the mountpath described in the pod definition)
 //filename - file to be read
 //namespace - optional param - namespace of the pod
 //Output  - k8s exec pod operation output and/or error
 func (rb *BlockOperation) BlockRead(name string, mountpath string, filename string, namespace string) (string, error) {
 	rd := mountpath + "/" + filename
-	readDataFromBlockPod[1] = rd
-	option := []string{name}
+	args := []string{"exec", name}
+
 	if namespace != "" {
-		option = append(option, namespace)
+		args = append(args, "-n", namespace)
 	}
-	out, err, status := rb.transportClient.Execute(readDataFromBlockPod, option)
-	if status == 0 {
-		return out, nil
+	args = append(args, "--", "cat", rd)
+
+	result, err := rb.k8sClient.Kubectl(args...)
```

Comment:
i would propose that no future checkins would call out to kubectl so we can transition to the api client